plugin/helpers_bots_test.go

```diff
@@ -151,5 +217,63 @@ func TestEnsureBot(t *testing.T) {
 			assert.Equal(t, "", botId)
 			assert.NotNil(t, err)
 		})
+
+		t.Run("should create bot and set the bot profile image when specified", func(t *testing.T) {
+			expectedBotId := model.NewId()
+
+			api := setupAPI()
+			api.On("KVGet", plugin.BOT_USER_KEY).Return([]byte(expectedBotId), nil)
+			api.On("GetUserByUsername", testbot.Username).Return(nil, nil)
+			api.On("CreateBot", testbot).Return(&model.Bot{
+				UserId: expectedBotId,
+			}, nil)
+			api.On("KVSet", plugin.BOT_USER_KEY, []byte(expectedBotId)).Return(nil)
+			api.On("GetBundlePath").Return("", nil)
+			defer api.AssertExpectations(t)
+
+			p := &plugin.HelpersImpl{}
+			p.API = api
+
+			testsDir, _ := fileutils.FindDir("tests")
+			testImage := filepath.Join(testsDir, "test.png")
+
+			imageBytes, err := ioutil.ReadFile(testImage)
+			assert.Nil(t, err)
+			api.On("SetProfileImage", expectedBotId, imageBytes).Return(nil)
+
+			botId, err := p.EnsureBot(testbot, plugin.ProfileImagePath(testImage))
+			api.AssertCalled(t, "SetProfileImage", expectedBotId, imageBytes)
+			assert.Equal(t, expectedBotId, botId)
+			assert.Nil(t, err)
+		})
+
+		t.Run("should create bot and set the bot icon image when specified", func(t *testing.T) {
```

Comment:
i think we're missing a case where the bot was never created, no?

Reply:
@ali-farooq0 This should have tested the case when the bot was never created, but I had a mistake in the test 🙈

Reply:
heh, no worries :)